daphne/src/roles_test.rs

```diff
@@ -1593,3 +1594,75 @@ async fn e2e_taskprov(version: DapVersion) {
 }
 
 async_test_version! { e2e_taskprov, Draft02 }
+
+fn early_metadata_checks(version: DapVersion) {
+    let t = Test::new(version);
+    let mut rng = thread_rng();
+    let metadata = ReportMetadata {
+        id: ReportId(rng.gen()),
+        time: t.now,
+        extensions: vec![],
+    };
+    // A current not processed, not collected report is OK!
+    assert_matches!(
+        early_metadata_check(&metadata, false, false, t.now - 100, t.now + 100),
```

Comment:
Readability: Variables improve understanding of the test without having to jump to the definition of `early_metadata_check`. 
```suggestion
        processed = false;
        collected = false;
        early_metadata_check(&metadata, processed, collected, t.now - 100, t.now + 100),
```